services/api.py

```diff
@@ -823,36 +823,34 @@ def get_category_ids(categories, category):
                     ids.append(mapping.split(':')[1])
             return ids
 
-        if 'category' in filters:
-            services_and_service_nodes = filters.get('category', None).split(',')
-            service_ids = get_category_ids(services_and_service_nodes, 'service')
-            servicenode_ids = get_category_ids(services_and_service_nodes, 'service_node')
+        categories = filters.get('category', None)
+        if categories is not None:
+            service_ids = get_category_ids(categories.split(','), 'service')
+            servicenode_ids = get_category_ids(categories.split(','), 'service_node')
             queryset = queryset.filter(Q(services__in=service_ids)
                                        | Q(service_nodes__in=service_nodes_by_ancestors(servicenode_ids))).distinct()
 
         def get_services_by_servicenodes(servicenode_ids):
-            ids = []
-            servicenodes = ServiceNode.objects.filter(id__in=service_nodes_by_ancestors(servicenode_ids))
-            for node in servicenodes:
-                servicelist = node.related_services.all()
-                ids.extend([service.id for service in servicelist])
+            ids = [service.id for service in
```

Comment:
You can replace the list comprehension with

    Service.objects.filter(...).values_list('id', flat=True)